### PR TITLE
fix(mongoose): add `virtuals: true` to the given options for json-mapper when a mongoose model is serialized

### LINKED ISSUE #1981: [BUG] VirtualRef unable to be mapped
## Information

- **Version:** 6.114.16
- Packages: @tsed/mongoose

Using VirtualRef, I am unable to respond with th model through the controller. 

Response does not contain "galaxies" or "galaxyCount": 

```js
{
  id: "62cffa39f8091984af6c55f8",
  name: "test",
  createdBy: {
    id: "62cffa39f8091984af6c55f9",
    name: "test",
  },
  dateCreation: "2022-07-14T11:12:57.955Z",
}
```

## Example

```ts
@Model()
export class GalaxiesModel {
  @ObjectID('id')
  @Groups('!internal')
  _id?: string;

  @Property()
  @Required()
  name: string;

  @Groups('!internal')
  spaceId?: string;

  @Groups('!internal')
  user?: UserInfo;

  @Format('date-time')
  @Groups('!internal')
  @Default(Date.now)
  @ReadOnly()
  dateCreation?: Date = new Date();
}
```

```ts
@Model()
export class SpacesModel {
  @ObjectID('id')
  @Groups('!internal')
  _id?: string;

  @Property()
  @Required()
  name: string;

  @Groups('!internal')
  createdBy?: UserInfo;

  @Groups('!internal')
  @VirtualRef({
    ref: GalaxiesModel,
    localField: '_id',
    foreignField: 'spaceId',
    justOne: false,
  })
  @CollectionOf(GalaxiesModel)
  galaxies?: VirtualRefs<GalaxiesModel>;

  @VirtualRef({
    ref: GalaxiesModel,
    localField: '_id',
    foreignField: 'spaceId',
    count: true,
  })
  @Groups('!internal')
  @Integer()
  galaxyCount?: number;

  @Format('date-time')
  @Groups('!internal')
  @Default(Date.now)
  @ReadOnly()
  dateCreation?: Date = new Date();
}
```

```ts
  @Get('/')
  async getSpaces(): Promise<{ spaces: SpacesModel[] }> {
    const spaces = await this.service.find(
      {},
      {},
      { populate: ['galaxies', 'galaxyCount'] }
    );

    return { spaces };
  }
```

image

## Acceptance criteria

- [ ] The response should contain the VirtualRef model (Case above, "galaxies" or "galaxyCount")

### PR DESCRIPTION
**Pull request by Romakita**
Closes: #1981

## Information

| Type                  | Breaking change |
| --------------------- | --------------- |
| Fix | No       |

---

## Todos

- [x] Tests
- [x] Coverage

### CODE REVIEW

File: packages/orm/mongoose/test/virtualRef.integration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+class SpacesController {
+  @Inject(SpacesModel)
+  spaces: MongooseModel<SpacesModel>;
+
+  @Inject(GalaxiesModel)
+  galaxies: MongooseModel<GalaxiesModel>;
+
+  @Get("/")
+  async getSpaces() {
+    return this.spaces.find({}, {}, {populate: ["galaxies", "galaxyCount"]});
```
Comment (Szauka) [inline packages/orm/mongoose/test/virtualRef.integration.spec.ts:62]: @Romakita Do we need to call populate like this?

Comment (Romakita) [reply packages/orm/mongoose/test/virtualRef.integration.spec.ts:62]: Not necessarly. It equivalent to use .populate()

**Pull request merged**